panoramapublic/src/org/labkey/panoramapublic/query/JournalManager.java

```diff
@@ -427,23 +429,25 @@ public static void removeJournalAccess(ExperimentAnnotations expAnnotations, Jou
     {
         JournalExperiment je = getJournalExperiment(expAnnotations, journal);
 
-        SimpleFilter filter = new SimpleFilter();
-        filter.addCondition(FieldKey.fromParts("JournalId"), journal.getId());
-        filter.addCondition(FieldKey.fromParts("ExperimentAnnotationsId"), expAnnotations.getId());
-        Table.delete(PanoramaPublicManager.getTableInfoJournalExperiment(), filter);
-
-        // Try to delete the short copy URL. Since we just deleted the entry in table JournalExperiment
-        // that references this URL we should not get a foreign key constraint error.
-        tryDeleteShortUrl(je.getShortCopyUrl(), user);
-        // Try to delete the short access URL only if the experiment has not yet been copied (accessURL points to journal's folder after copy)
-        // OR the access url is no longer referenced in the ExperimentAnnotations table.
-        if(je.getCopied() == null || ExperimentAnnotationsManager.getExperimentForShortUrl(je.getShortAccessUrl()) == null)
+        if(je.getJournalExperimentId() == null)
         {
-            tryDeleteShortUrl(je.getShortAccessUrl(), user);
+            // This experiment has not yet been copied to Panorama Public so we can delete the row in JournalExperiment
+            SimpleFilter filter = new SimpleFilter();
```

Comment:
Does any upstream code validate that the supplied IDs belong to a container that the user has permission to operate on?

Reply:
Yes. The action requires AdminPermission, and there is an ensureCorrectContainer() method that redirects to the container that the ID belongs to.   I did a quick review of all the actions in PanoramaPublicController and added the container check in two places where it wasn't being done.